source/dofs/dof_tools.cc

```diff
@@ -1419,7 +1419,7 @@ namespace DoFTools
     // If we have a distributed::Triangulation only allow locally_owned
     // subdomain.
     Assert (
-      (dof_handler.get_triangulation().locally_owned_subdomain() == numbers::invalid_subdomain_id)
+      (dof_handler.get_triangulation().locally_owned_subdomain() == 0)
```

Comment:
looking at the function, i think this one is probably meant to test that either `tria` is serial (`subdomain` can be whatever) or `subdomain` matches that stored in parallel tria.